djpsa/halo/views.py

```diff
@@ -43,24 +42,41 @@ def post(self, request, *args, **kwargs):
             logger.error('Error decoding JSON for callback: %s', e)
             return HttpResponse(status=400)
 
-        sync = None
+        self.handle(data)
 
-        for record_type, sync_class in self.CALLBACK_TYPES.items():
-            # Searches for record type keys in data, i.e. if it's a ticket
-            # callback, it will find 'ticket'.
-            if record_type in data:
-                sync = sync_class()
+        return HttpResponse(status=200)
 
-                instance, _ = sync.update_or_create_instance(
-                    data.get(record_type))
+    def handle(self, data):
+        """
+        Do the interesting stuff here, so that it can be overridden in
+        a child class if needed.
+        """
 
-                # Sync related records, actions, appointments, etc.
-                sync.sync_related(instance)
-                break
+        if self.callback_handler:
+            # Call the callback handler if it's defined
+            self.callback_handler(data)
+        else:
+            sync = self.sync_class()
 
-        if sync is None:
-            # This is normal if we get a callback for a record type we don't
-            # care about somehow.
-            return HttpResponse(status=400)
+            instance, _ = sync.update_or_create_instance(
+                data.get(self.entity_type))
 
-        return HttpResponse(status=200)
+            # Sync related records, actions, appointments, etc.
+            sync.sync_related(instance)
+
+
+class TicketCallBackView(CallBackView):
+
+    entity_type = 'ticket'
+    sync_class = TicketSynchronizer
+
+    # This must be defined on child classes or the parent class
+    # handler will be overridden for each child class.
+    callback_handler = None
+
+    @classmethod
+    def register_callback_handler(cls, callback_handler):
```

Comment:
Should this be on TicketCallBackView or on CallBackView? At such time there are other callback handlers in Halo, I would expect them to need this functionality too.

Reply:
Oh yeah, it can, must have slipped it on the child by mistake